### PR TITLE
feat(content): Added creating content item from template.

### PR DESCRIPTION
**Pull request by eozan**
SD-2236

### CODE REVIEW

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 1]
```diff
@@ -1203,9 +1203,23 @@
...
-    function ContentCreateDirective() {
+    ContentCreateDirective.$inject = ['templatesSettingsService'];
+    function ContentCreateDirective(templatesSettingsService) {
         return {
-            templateUrl: 'scripts/superdesk-authoring/views/sd-content-create.html'
+            templateUrl: 'scripts/superdesk-authoring/views/sd-content-create.html',
+            link: function(scope) {
+                scope.contentTemplates = null;
+
+                var fetchTemplates = function() {
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: We can get rid of this private function here, as its only purpose is to be invoked once when the directive is linked. Just execute the function's body and that's it. :wink:

File: client/app/scripts/superdesk-workspace/content/content.js [hunk 2]
```diff
@@ -4,6 +4,20 @@
 ContentCtrlFactory.$inject = ['api', 'superdesk'];
 function ContentCtrlFactory(api, superdesk) {
     return function ContentCtrl($scope) {
```
Comment (plamut) [inline client/app/scripts/superdesk-workspace/content/content.js]: `$scope` is not used anywhere, we can remove it as a dependency.

**Comment by plamut** [comment]
Apart from a few minor remarks it seems good. But please add unit tests for the new code before merging.

**Comment by ioanpocol** [comment]
An e2e tests will help.

**Comment by ioanpocol** [comment]
:+1:

**Comment by plamut** [comment]
IMO e2e tests should not be viewed as a replacement for unit tests. e2e tests are generally slow to run and it's not feasible tu run them after every code change (i.e. when you are about to commit something).

**Comment by eozan** [comment]
unit tests added.

**Comment by plamut** [comment]
That's good!

:+1:

**Comment by amagdas** [comment]
:+1:

**Pull request merged**